### PR TITLE
APP-1876 : convert proposalId from bigInt to bytes32

### PR DESCRIPTION
**Pull request by Rekard0**
## Description

Task: [APP-1876](https://aragonassociation.atlassian.net/browse/APP-1876)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-1876]: https://aragonassociation.atlassian.net/browse/APP-1876?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/subgraph/src/packages/token/token-voting.ts [hunk 1]
```diff
@@ -21,7 +26,14 @@ import {
 
 import {RATIO_BASE, VOTER_OPTIONS, VOTING_MODES} from '../../utils/constants';
 import {fetchERC20} from '../../utils/tokens/erc20';
+import {bigIntToBytes32} from '../../utils/bytes';
 
+function generateProposalId(plugin: Address, pluginProposalId: BigInt): string {
```
Comment (novaknole) [inline packages/subgraph/src/packages/token/token-voting.ts:31]: well, you already got this. why not using it in other files ? and put it in the utils somewhere.

better name: `getProposalId`.

File: packages/subgraph/src/packages/token/token-voting.ts [hunk 2]
```diff
@@ -78,7 +90,7 @@ export function _handleProposalCreated(
       let actionId =
         event.address.toHexString() +
```
Comment (novaknole) [inline packages/subgraph/src/packages/token/token-voting.ts:91]: you can use generated proposalId here.

File: packages/subgraph/src/plugin/pluginSetupProcessor.ts [hunk 3]
```diff
@@ -96,13 +98,13 @@ export function handleInstallationApplied(event: InstallationApplied): void {
...
   }
-  pluginEntity.pluginAddress =  event.params.plugin;
+  pluginEntity.pluginAddress = event.params.plugin;
   pluginEntity.appliedPreparation = preparationId;
   pluginEntity.appliedSetupId = event.params.appliedSetupId;
   pluginEntity.state = 'Installed';
   pluginEntity.save();
 
-  addPlugin(dao, event.params.plugin);
+  // addPlugin(dao, event.params.plugin);
```
Comment (novaknole) [inline packages/subgraph/src/plugin/pluginSetupProcessor.ts:107]: why commented

File: packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts [hunk 4]
```diff
@@ -111,7 +112,7 @@ test('Run AddresslistVoting (handleProposalCreated) mappings with mock event', (
   let entityID =
     Address.fromString(CONTRACT_ADDRESS).toHexString() +
     '_' +
-    BigInt.fromString(proposalId).toHexString();
+    bigIntToBytes32(BigInt.fromString(proposalId));
```
Comment (novaknole) [inline packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts:115]: thats the beauty now, in tests you should be using `getProposalId`.

File: packages/subgraph/tests/constants.ts [hunk 5]
```diff
@@ -18,7 +19,9 @@ export const THREE = '3';
 
 export const PROPOSAL_ID = ZERO;
 export const PROPOSAL_ENTITY_ID =
-  Address.fromString(CONTRACT_ADDRESS).toHexString() + '_0x' + PROPOSAL_ID;
+  Address.fromString(CONTRACT_ADDRESS).toHexString() +
```
Comment (novaknole) [inline packages/subgraph/tests/constants.ts:22]: getProposalId.

File: packages/subgraph/tests/constants.ts [hunk 6]
```diff
@@ -50,6 +53,16 @@ export const CREATED_AT = '1644850000';
...
 export const ZERO_BYTES32 =
   '0x0000000000000000000000000000000000000000000000000000000000000000';
+export const ONE_BYTES32 =
+  '0x0000000000000000000000000000000000000000000000000000000000000001';
+export const HALF_UINT256_BYTES32 =
+  '0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
+export const MAX_UINT256_BYTES32 =
+  '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
+
+export const MAX_UINT256_NUMBER_STRING =
```
Comment (novaknole) [inline packages/subgraph/tests/constants.ts:63]: why not having it as `BigInt` type here ? why string ? up to you. I don't know the exact thing.

**Comment by novaknole** [comment]
Generally, I don't get why you have duplicated code mostly for the proposal Id generation. why not using `getProposalId` everywhere ?

File: packages/subgraph/src/packages/token/token-voting.ts (hunk 2 above)
Comment (Rekard0) [reply packages/subgraph/src/packages/token/token-voting.ts:81]: thanks, nice catch.

File: packages/subgraph/tests/constants.ts (hunk 5 above)
Comment (Rekard0) [reply packages/subgraph/tests/constants.ts:22]: thanks, amended

File: packages/subgraph/tests/constants.ts (hunk 6 above)
Comment (Rekard0) [reply packages/subgraph/tests/constants.ts:63]: I want to create all bigInts from string and then check the function `bigIntToBytes32` result against expected result, for testing purposes

File: packages/subgraph/src/packages/token/token-voting.ts (hunk 1 above)
Comment (Rekard0) [reply packages/subgraph/src/packages/token/token-voting.ts]: outdated

Comment (Rekard0) [reply packages/subgraph/src/packages/token/token-voting.ts]: but amended

File: packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts (hunk 4 above)
Comment (Rekard0) [reply packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts]: outdated, but switched to import `PROPOSAL_ENTITY_ID`

File: packages/subgraph/src/packages/addresslist/addresslist-voting.ts [hunk 7]
```diff
@@ -107,10 +107,11 @@ export function _handleProposalCreated(
 }
 
 export function handleVoteCast(event: VoteCast): void {
+  const pluginProposalId = event.params.proposalId;
   const member = event.params.voter.toHexString();
   const pluginId = event.address.toHexString();
   const memberId = pluginId + '_' + member;
-  let proposalId = pluginId + '_' + event.params.proposalId.toHexString();
+  let proposalId = pluginId + '_' + bigIntToBytes32(pluginProposalId);
```
Comment (novaknole) [inline packages/subgraph/src/packages/addresslist/addresslist-voting.ts:114]: `generateProposalId`.

File: packages/subgraph/src/packages/token/token-voting.ts [hunk 8]
```diff
@@ -76,9 +78,7 @@ export function _handleProposalCreated(
       const action = actions[index];
 
       let actionId =
-        event.address.toHexString() +
-        '_' +
-        event.params.proposalId.toHexString() +
+        generateProposalId(event.address, pluginProposalId) +
         '_' +
```
Comment (novaknole) [inline packages/subgraph/src/packages/token/token-voting.ts:82]: Just commenting here, but let's start using `concat` from now on. In order not to make the PR huge, only use `concat` for things that you did in the PR anyways. in (`getProposal`, for `actionId`)

File: packages/subgraph/src/packages/addresslist/addresslist-voting.ts (hunk 7 above)
Comment (Rekard0) [reply packages/subgraph/src/packages/addresslist/addresslist-voting.ts:114]: amended

File: packages/subgraph/src/packages/token/token-voting.ts (hunk 8 above)
Comment (Rekard0) [reply packages/subgraph/src/packages/token/token-voting.ts:82]: amended

File: packages/subgraph/src/plugin/pluginSetupProcessor.ts (hunk 3 above)
Comment (Rekard0) [reply packages/subgraph/src/plugin/pluginSetupProcessor.ts]: we moved to prepare installation, as it was not picking up plugin settings
will be fixed in future pr, 
it might need more refactoring

File: packages/subgraph/src/packages/token/token-voting.ts [hunk 9]
```diff
@@ -75,12 +76,9 @@ export function _handleProposalCreated(
     for (let index = 0; index < actions.length; index++) {
       const action = actions[index];
 
-      let actionId =
-        event.address.toHexString() +
-        '_' +
-        event.params.proposalId.toHexString() +
-        '_' +
-        index.toString();
+      let actionId = getProposalId(event.address, pluginProposalId).concat(
```
Comment (novaknole) [inline packages/subgraph/src/packages/token/token-voting.ts:79]: 1. `.concat('_')` missing.
2. why do tests succeed ?

File: packages/subgraph/tests/admin/admin.test.ts [hunk 10]
```diff
@@ -105,28 +106,31 @@ test('Run Admin plugin (handleProposalExecuted) mappings with mock event', () =>
...
   adminProposal.createdAt = BigInt.fromString(START_DATE);
   adminProposal.startDate = BigInt.fromString(START_DATE);
   adminProposal.endDate = BigInt.fromString(START_DATE);
   adminProposal.allowFailureMap = BigInt.fromString(ALLOW_FAILURE_MAP);
-  adminProposal.administrator = adminstratorAddress.toHexString();
+  adminProposal.administrator = administratorAddress.toHexString();
   adminProposal.save();
 
-  const actionId = CONTRACT_ADDRESS + '_' + PROPOSAL_ID + '_' + PROPOSAL_ID;
+  const actionId = CONTRACT_ADDRESS.concat('_')
```
Comment (novaknole) [inline packages/subgraph/tests/admin/admin.test.ts:130]: why not `getProposalId`  or `PROPOSAL_ENTITY_ID` ?

File: packages/subgraph/tests/utils/utils.test.ts [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  HALF_UINT256_BYTES32,
+  MAX_UINT256_BYTES32
+} from '../constants';
+import {TestCase} from './utils';
+
+describe('Test bytes', function() {
+  test('`bigIntToBytes32` with a range of `bigInt`s', function() {
+    const MAX_UINT256 = BigInt.fromString(MAX_UINT256_NUMBER_STRING);
+
+    const testCases: TestCase[] = [
```
Comment (novaknole) [inline packages/subgraph/tests/utils/utils.test.ts:21]: about the `TestCase` ? do we have alternative ? This could be an overkill.

1. Would love to see if it's a recommended practice and in the long run, it won't. hurt us instead of helping us.
2. Depending on answer 1, if we have to have it, then `TestCase` might be a wrong name. What if I wanna create another TestCase which instead of `BigInt`, expects `String` as an argument. Thats why you will need to be specific to name it `TestCaseInputBigInt` and this is where It might start to become an overkill.

File: packages/subgraph/tests/admin/admin.test.ts [hunk 12]
```diff
@@ -105,28 +106,31 @@ test('Run Admin plugin (handleProposalExecuted) mappings with mock event', () =>
...
+    .concat(PROPOSAL_ID)
+    .concat('_')
+    .concat(PROPOSAL_ID);
```
Comment (mathewmeconry) [inline packages/subgraph/tests/admin/admin.test.ts:133]: shouldn't this be the index of the action in the actions array?

File: packages/subgraph/tests/utils/utils.test.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+  MAX_UINT256_NUMBER_STRING,
+  ZERO_BYTES32,
+  ONE_BYTES32,
+  HALF_UINT256_BYTES32,
+  MAX_UINT256_BYTES32
+} from '../constants';
+import {TestCase} from './utils';
+
+describe('Test bytes', function() {
+  test('`bigIntToBytes32` with a range of `bigInt`s', function() {
```
Comment (mathewmeconry) [inline packages/subgraph/tests/utils/utils.test.ts:18]: Personally, I prefer having one test case per test. Makes it sometimes easier to debug and finding some weird edge case bugs

File: packages/subgraph/src/packages/token/token-voting.ts (hunk 9 above)
Comment (Rekard0) [reply packages/subgraph/src/packages/token/token-voting.ts:79]: amended

File: packages/subgraph/tests/admin/admin.test.ts (hunk 12 above)
Comment (novaknole) [reply packages/subgraph/tests/admin/admin.test.ts:133]: It seems to me that because of these, tests are unstable. It needs to be written better or completely removed. Dont see any other point.

Comment (Rekard0) [reply packages/subgraph/tests/admin/admin.test.ts:133]: yes, amended

File: packages/subgraph/tests/admin/admin.test.ts (hunk 10 above)
Comment (Rekard0) [reply packages/subgraph/tests/admin/admin.test.ts:130]: amended

File: packages/subgraph/tests/utils/utils.test.ts (hunk 13 above)
Comment (Rekard0) [reply packages/subgraph/tests/utils/utils.test.ts:18]: amended

**Review by mathewmeconry — APPROVED** [review]

**Review by novaknole — APPROVED** [review]
